net_handler/udp_conn.c

```diff
@@ -197,36 +197,39 @@ static void* send_device_data(void* args) {
 }
 
 
-static void* update_gamepad_state(void* args) {
-    static int size = sizeof(GpState);
+static void* update_inputs(void* args) {
+    static int size = 128;
     uint8_t buffer[size];
     int recvlen;
 
     while (1) {
         recvlen = recvfrom(socket_fd, buffer, size, 0, (struct sockaddr*) &dawn_addr, &addr_len);
         // log_printf(DEBUG, "Dawn IP is %s:%d", inet_ntoa(dawn_addr.sin_addr), ntohs(dawn_addr.sin_port));
         if (recvlen == size) {
-            log_printf(WARN, "update_gamepad_state: UDP Read length matches read buffer size %d", recvlen);
+            log_printf(WARN, "update_inputs: UDP Read length matches read buffer size %d", recvlen);
         }
         if (recvlen < 0) {
-            log_printf(ERROR, "update_gamepad_state: UDP recvfrom failed: %s", strerror(errno));
+            log_printf(ERROR, "update_inputs: UDP recvfrom failed: %s", strerror(errno));
             continue;
         }
-        GpState* gp_state = gp_state__unpack(NULL, recvlen, buffer);
-        if (gp_state == NULL) {
-            log_printf(ERROR, "update_gamepad_state: Failed to unpack GpState");
+        UserInputs* inputs = user_inputs__unpack(NULL, recvlen, buffer);
+        if (inputs == NULL) {
+            log_printf(ERROR, "update_inputs: Failed to unpack UserInputs");
             continue;
         }
-
-        robot_desc_write(GAMEPAD, gp_state->connected ? CONNECTED : DISCONNECTED);
-        if (gp_state->connected) {
-            if (gp_state->n_axes != 4) {
-                log_printf(ERROR, "update_gamepad_state: Number of joystick axes given is %d which is not 4. Cannot update gamepad state", gp_state->n_axes);
-            } else {
-                gamepad_write(gp_state->buttons, gp_state->axes);
+        for (int i = 0; i < inputs->n_inputs; i++) {
+            Input* input = inputs->inputs[i];
+            robot_desc_field_t source = GAMEPAD + input->source;  // Assumes that order in enum is GAMEPAD, KEYBOARD
```

Comment:
So `input->source` is an int right? So if the `input->source` is GAMEPAD, the small addition here is just 0(GAMEPAD) + 0(input->source)?

Reply:
Ah never mind, just realized youre iterating over inputs which cuases input to change each time

Reply:
I didn't comment on this enum thing because I wasn't sure what the best solution was, but what Daniel and Ben mentioned above sounds good to me!

Reply:
Hmm ok I wanted to make it general but a ternary condition would be cleaner if we are sticking with only 2 sources.

The main problem is `input->source` is a Protobuf enum corresponding to 0 and 1. So I need to convert the protobuf enum to a Runtime robot_desc enum. I do this in several places and so didn't want to use a if/switch case.